docs/pvc-recovery.md

```diff
@@ -0,0 +1,263 @@
+```
+    - e.g.
+```
+    gateway:
+      recover_after_nodes: 2
+      expected_nodes: 3
+      recover_after_time: ${RECOVER_AFTER_TIME}
+
+    path:
+      data: /elasticsearch/persistent/${CLUSTER_NAME}/data
+      logs: /elasticsearch/persistent/${CLUSTER_NAME}/logs
+
+    reindex:
+      remote.whitelist: "elasticsearch-recovery.openshift-logging.svc:9200"
+      ssl:
+        verification_mode: certificate
+        truststore:
+          path: /etc/elasticsearch/secret/truststore
+          password: tspass
+          type: jks
+        keystore:
+          path: /etc/elasticsearch/secret/key
+          type: jks
+          password: kspass
+```
+
+1. Restart the cluster to pick up the configmap changes:
+  - `oc delete pod -l cluster-name=elasticsearch`
+
+1. Wait for the cluster to come back up
+
+1. Run the recovery script (in the appendix) from your local machine where you are running your other `oc` commands.
+
+1. Double-check your existing cluster's indices to ensure old data had been recovered.
+  - Indices will have "recovered" in the name
+
+1. If required to recover other volumes, scale down the recovery cluster and repeat steps 9 - 12 using the other PVC and rerun the recovery script.
+
+1. As part of clean up, remove the old unused PVC that data has been recovered from
+  - `oc delete pvc <pvc_name>`
+
+1. Remove the elasticsearch recovery CR
+  - `oc delete elasticsearch elasticsearch-recovery`
+
+1. Update the elasticsearch CR to be `Managed` again.
+
+## Appendix
+
+### Example Recovery CR
+
+```
+apiVersion: logging.openshift.io/v1
+kind: Elasticsearch
+metadata:
+  name: elasticsearch-recovery
+  namespace: openshift-logging
+spec:
+  managementState: Managed
+  nodeSpec:
+    proxyResources:
+      limits:
+        memory: 64Mi
+      requests:
+        cpu: 100m
+        memory: 64Mi
+    resources:
+      limits:
+        memory: 1Gi
+      requests:
+        cpu: 150m
+        memory: 1Gi
+  nodes:
+  - nodeCount: 3
+    proxyResources: {}
+    resources: {}
+    roles:
+    - client
+    - data
+    - master
+    storage: {}
+  redundancyPolicy: SingleRedundancy
+  ```
+
+
+### Recovery script
+
+```
+#!/bin/bash
+
+recovery_node=$1
+destination_node=$2
+
+function seed_recovery_templates() {
+
+  kibana=$(oc exec -c elasticsearch $destination_node -- es_util --query="_template/common.settings.kibana.template.json?pretty")
+
+  primary="$(echo $kibana | jq -r '.["common.settings.kibana.template.json"].settings.index.number_of_shards')"
+  replicas="$(echo $kibana | jq -r '.["common.settings.kibana.template.json"].settings.index.number_of_replicas')"
+
+operations=$(cat <<EOF
+{
+  "order": 0,
+  "index_patterns": [
+    ".operations.*"
+  ],
+  "settings": {
+    "index": {
+      "number_of_shards": $primary,
+      "number_of_replicas": $replicas
+    }
+  }
+}
+EOF
+)
+
+  oc exec -c elasticsearch $destination_node -- es_util --query="_template/recovery.operations.template.json" -XPUT -d "${operations}"
```

Comment:
Are we deleting this template after we are done?

Reply:
no, but it shouldn't matter too much right? if a customer had upgraded their cluster they would have old templates still

Reply:
These recovery index templates are for the old data model only, right? If so then yes, we can probably keep it there (although a one time -X DELETE command in the end should be difficult to execute in the end, WDYT?)

Reply:
true, we could delete them after... i'll add that to the script example